packages/client/hmi-client/src/services/data.ts

```diff
@@ -399,59 +399,68 @@ const searchXDDArticles = async (term: string, xddSearchParam?: XDDSearchParams)
 	//  with a scan-and-scroll cursor that allows client to step through all results page-by-page.
 	//  NOTE: the "max" parameter will be ignored
 	//  NOTE: results may not be ranked in this mode
-	let url = `/xdd/documents?term=${term}`;
+	let searchParams = `term=${term}`;
+	const url = '/xdd/documents?';
 
 	if (xddSearchParam?.docid) {
-		url += `&docid=${xddSearchParam.docid}`;
+		searchParams += `&docid=${xddSearchParam.docid}`;
 	}
 	if (xddSearchParam?.doi) {
-		url += `&doi=${xddSearchParam.doi}`;
-	}
-	if (xddSearchParam?.title) {
-		url += `&title=${xddSearchParam.title}`;
+		searchParams += `&doi=${xddSearchParam.doi}`;
 	}
 	if (xddSearchParam?.dataset) {
-		url += `&dataset=${xddSearchParam.dataset}`;
+		searchParams += `&dataset=${xddSearchParam.dataset}`;
 	}
 	if (xddSearchParam?.dict && xddSearchParam?.dict.length > 0) {
-		url += `&dict=${xddSearchParam.dict.join(',')}`;
+		searchParams += `&dict=${xddSearchParam.dict.join(',')}`;
 	}
 	if (xddSearchParam?.min_published) {
-		url += `&min_published=${xddSearchParam.min_published}`;
+		searchParams += `&min_published=${xddSearchParam.min_published}`;
 	}
 	if (xddSearchParam?.max_published) {
-		url += `&max_published=${xddSearchParam.max_published}`;
+		searchParams += `&max_published=${xddSearchParam.max_published}`;
 	}
 	if (xddSearchParam?.pubname) {
-		url += `&pubname=${xddSearchParam.pubname}`;
+		searchParams += `&pubname=${xddSearchParam.pubname}`;
 	}
 	if (xddSearchParam?.publisher) {
-		url += `&publisher=${xddSearchParam.publisher}`;
+		searchParams += `&publisher=${xddSearchParam.publisher}`;
 	}
 	if (xddSearchParam?.includeHighlights) {
-		url += '&include_highlights=true';
+		searchParams += '&include_highlights=true';
 	}
 	if (xddSearchParam?.inclusive) {
-		url += '&inclusive=true';
+		searchParams += '&inclusive=true';
 	}
 	if (enablePagination) {
-		url += '&full_results';
+		searchParams += '&full_results';
 	} else {
 		// request results to be ranked
-		url += '&include_score=true';
+		searchParams += '&include_score=true';
 	}
 	if (xddSearchParam?.facets) {
-		url += '&facets=true';
+		searchParams += '&facets=true';
+	}
+
+	// search title and abstract when performing term-based search if requested
+	if (term !== '' && xddSearchParam?.additional_fields) {
```

Comment:
Is there ever a risk of `term` being Undefined/null/etc?  If there is i believe that this will evaluate to `true`.

Reply:
term is defined as string and should be a valid string even if empty. I don't think term will ever be undefined/null